### DISCUSSION TITLE
Not able to clone any of my repo to local

### DISCUSSION THREAD

**Discussion by arunraj3** [community/community#60658] [Repositories] [answered]
### Select Topic Area

Question

### Body

I have checked all the requirements to clone the repo 
1. For easier cloning made the repo public 
2. Checked the internet connection

**Comment by leonden** [comment]
How did you clone the repo?
On your local machine, make sure that Git is installed and then you can run

```sh
git clone https://github.com/arunraj3/<YourRepo>.git
```

though I assume, that you already did that.

The visibility of your repo shouldn't affect the cloning at all.
Can you provide further information to the actual issue?

**Reply by arunraj3** [reply]
Ya I have installed git.
And I have synced my Account with Vscode 
But I was able to pull and push the repo 
But I not able to clone it 
Is pulling the repo is same as cloning?
While cloning is it showing
"fatal error"

**Reply by leonden** [reply]
If you pull on your local copy of the remote repository, you basically synchronize the state of the code. You can use _git pull_ to get new code that is already pushed to the remote repository from another contributor.
Can you show me the actual error shown in your shell?

**Reply by arunraj3** [reply]
fatal: '=' does not appear to be a git repository
fatal: Could not read from remote repository.      

Please make sure you have the correct access rights
and that the repository exists.

This is the error I'm getting

**Reply by leonden** [reply, marked as answer]
You could start by checking if the origin is set. simply run

```sh
git remote -v
```

This should show you all of the push / fetch remotes for your repository.

If this returns with no output, please go directly to the last code block.

Then verify the remote's name and its adress.

If this returns showing that you have remotes set, check that the name of the remote matches the remote you are using in your commands.

```sh
git remote -v

YourOrigin ssh://git@example.com:1234/YourRepo.git (fetch)
YourOrigin ssh://git@example.com:1234/YourRepo.git (push)

# this should fail because `origin` is not set
git push origin main

# you need to use
git push YourOrigin main
```

You could rename the remote or change its URL if you want. Simply remove the old and then add the new correct remote.

```sh
git remote remove YourOrigin
```

Lastly you can add the proper remote with

```sh
git remote add origin ssh://git@example.com:1234/YourRepo.git

# this should now work as expected
git push origin main
```

Hope that helps! :)

**Reply by arunraj3** [reply]
Thanks for helping me, Leonden😊

**Reply by leonden** [reply]
No worries! I'm glad that I was able to help you 😄 
Happy coding 👨‍💻

**Comment by arunraj3** [comment]
Hello Leonden,
I'm very much new to open source how can I start to contribute to open source projects, it's overwhelming for me like where to start 
could you suggest to me where I can get started

**Reply by leonden** [reply]
Hi @arunraj3 
When people ask me that particular question, I always refer to [this repository](https://github.com/firstcontributions/first-contributions). It shows and explaines the exact and detailed steps you need for controbuting the first time in an open source project. They are most likely better than I am at explaining but if you have any questions left though, please do not hesitate to ask :smile:
